src/tls_wrap.cc

```diff
@@ -836,6 +840,156 @@ int TLSWrap::SelectSNIContextCallback(SSL* s, int* ad, void* arg) {
   return SSL_TLSEXT_ERR_OK;
 }
 
+#ifndef OPENSSL_NO_PSK
+
+void TLSWrap::SetPskIdentityHint(const FunctionCallbackInfo<Value>& args) {
+  TLSWrap* p;
+  ASSIGN_OR_RETURN_UNWRAP(&p, args.Holder());
+  CHECK_NOT_NULL(p->ssl_);
+
+  Environment* env = p->env();
+
+  CHECK(args[0]->IsArrayBufferView());
+  char* buf = Buffer::Data(args[0]);
+  size_t buflen = Buffer::Length(args[0]);
+
+  char* hint = Malloc(buflen + 1);
+  memcpy(hint, buf, buflen);
+  hint[buflen] = 0;
+
+  if (!SSL_use_psk_identity_hint(p->ssl_.get(), hint)) {
```

Comment:
The hint is a Buffer in the API, which implies (to me) that it can be arbitrary binary data, but the ossl API constrains it to be NUL terminated string, so perhaps the API should have the hint as a `<string>` not a Buffer?

Reply:
Well, it's not specified by the [spec](https://tools.ietf.org/html/rfc4279#section-5.2) what the identity hint should be though openssl requires in to be NUL terminated (https://www.openssl.org/docs/man1.1.1/man3/SSL_use_psk_identity_hint.html) hence the confusion. I personally think Buffer is better here as we make sure to not misinterpret the user data (using i.e. utf8 encoding).

Reply:
The only way to use strings (from what I understand) is to use latin1 encoding. Do you think it will be better than buffer?

Reply:
Given that the identity is constrained to be a [UTF-8 string](https://tools.ietf.org/html/rfc4279#section-5.1), and the only uses for the hint I can [find](https://stackoverflow.com/questions/8678199/psk-hint-with-openssl/8753393#8753393) use simple strings, and at least one other API uses [UTF-8](https://docs.oracle.com/javame/8.3/api/ssl-extension/api/com/oracle/ssl/PSKCredentials.html) I don't see a problem with node constraining the identity hint to be a UTF-8 string, too.

Reply:
I also think it's a good idea (this will make them easier to use and make api more consistent). Though Ben also mentioned this part (when it was still latin1 string https://github.com/nodejs/node/pull/23188#discussion_r221917732) so, @bnoordhuis wdyt?

Reply:
its now a string :+1: